converter/converters/pod/affinity.go

```diff
@@ -0,0 +1,398 @@
+package pod
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"github.com/koki/short/types"
+)
+
+func RevertAffinities(a []types.Affinity) (*v1.Affinity, error) {
+	node, pod, antiPod, err := splitAffinities(a)
+	if err != nil {
+		return nil, err
+	}
+
+	n, err := nodeAffinity(node)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := podAffinity(pod)
+	if err != nil {
+		return nil, err
+	}
+
+	ap, err := podAntiAffinity(antiPod)
+	if err != nil {
+		return nil, err
+	}
+
+	if n == nil && p == nil && ap == nil {
+		return nil, nil
+	}
+
+	return &v1.Affinity{
+		NodeAffinity:    n,
+		PodAffinity:     p,
+		PodAntiAffinity: ap,
+	}, nil
+}
+
+type PodAffinity struct {
+	Affinity   string
+	Topology   string
+	Namespaces []string
+}
+
+func splitAffinities(as []types.Affinity) (node []string, pod, antiPod []PodAffinity, err error) {
+	node = []string{}
+	pod = []PodAffinity{}
+	antiPod = []PodAffinity{}
+
+	for _, a := range as {
+		switch {
+		case len(a.NodeAffinity) > 0:
+			node = append(node, a.NodeAffinity)
+		case len(a.PodAffinity) > 0:
+			pod = append(pod, PodAffinity{
+				Affinity:   a.PodAffinity,
+				Topology:   a.Topology,
+				Namespaces: a.Namespaces,
+			})
+		case len(a.PodAntiAffinity) > 0:
+			antiPod = append(antiPod, PodAffinity{
+				Affinity:   a.PodAntiAffinity,
+				Topology:   a.Topology,
+				Namespaces: a.Namespaces,
+			})
+		default:
+			err = fmt.Errorf("unrecognized affinity %#v", a)
+		}
+	}
+
+	return
+}
+
+func podAntiAffinity(as []PodAffinity) (*v1.PodAntiAffinity, error) {
+	if len(as) == 0 {
+		return nil, nil
+	}
+
+	hard, soft, err := podAffinities(as)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(hard) == 0 {
+		hard = nil
+	}
+
+	if len(soft) == 0 {
+		soft = nil
+	}
+
+	return &v1.PodAntiAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution:  hard,
+		PreferredDuringSchedulingIgnoredDuringExecution: soft,
+	}, nil
+}
+
+func podAffinity(as []PodAffinity) (*v1.PodAffinity, error) {
+	if len(as) == 0 {
+		return nil, nil
+	}
+
+	hard, soft, err := podAffinities(as)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(hard) == 0 {
+		hard = nil
+	}
+
+	if len(soft) == 0 {
+		soft = nil
+	}
+
+	return &v1.PodAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution:  hard,
+		PreferredDuringSchedulingIgnoredDuringExecution: soft,
+	}, nil
+}
+
+func podAffinities(as []PodAffinity) (hard []v1.PodAffinityTerm, soft []v1.WeightedPodAffinityTerm, err error) {
```

Comment:
Wait, how is this different from podAffinity? I don't understand the naming here 

Also, you've used named return values.

Reply:
renamed it to splitAndRevertPodAffinity because it splits the affinities into hard/soft and also reverts them to native k8s types.